pyrtools/tools/display.py

```diff
@@ -474,27 +499,84 @@ def imshow(image, vrange='indep1', zoom=1, title='', col_wrap=None, ax=None,
         else:
             image_tmp.append(np.array(img))
             title_tmp.append(t)
-    image = np.array(image_tmp)
-    title = title_tmp
+    try:
+        image_tmp = np.array(image_tmp)
+    except ValueError:
+        # this happens when the images are the same shape but at least one is
+        # RGB(A) and at least one is grayscale, e.g., image_tmp[0].shape = (10,
+        # 10) and image_tmp[1].shape = (10, 10, 4). Strangely enough, it's not
+        # a problem if, in the above example, image_tmp[1].shape = (5, 5, 4).
+        # So in this case, we need to add extra dims, adding the RGB and/or A
+        # channels.
+        for i, img in enumerate(image_tmp):
+            if img.ndim == 2:
+                # make all of the RGB channels identical so it's grayscale
+                img = np.dstack([img, img, img])
+            if img.shape[-1] == 3:
+                # add an alpha channel of all 1s
+                img = np.dstack([img, np.ones_like(img[..., 0])])
+            image_tmp[i] = img
+        image_tmp = np.array(image_tmp)
+    return image_tmp, title_tmp, contains_rgb
```

Comment:
This bit here is to support mixing grayscale, RGB, and RGBA images. I couldn't find a way to create an array containing an array of size, e.g., `(10, 10)` and `(10, 10, 4)` (though `(5,5)` and `(10,10,4)` is okay), so I expand the grayscale image into an RGBA image in order to make this work. I just duplicate the grayscale values across all three channels -- that is correct, right? That's what `skimage.color.gray2rgb` does (I don't use it because we don't have `skimage` as a requirement).